File: ganilib-core/src/main/java/com/gani/lib/ui/view/GAutoCompleteTextView.java
package com.gani.lib.ui.view;

import android.content.Context;
import android.support.v7.widget.AppCompatAutoCompleteTextView;
import android.util.AttributeSet;

public class GAutoCompleteTextView<T extends GAutoCompleteTextView> extends AppCompatAutoCompleteTextView implements IView {
  private ViewHelper helper;

  public GAutoCompleteTextView(Context context) {
    super(context);
    init();
  }

  public GAutoCompleteTextView(Context context, AttributeSet attrs) {
    super(context, attrs);
    init();
  }

  private void init() {
    this.helper = new ViewHelper(this);
  }

  private T self() {
    return (T) this;
  }

  @Override
  public GAutoCompleteTextView padding(Integer left, Integer top, Integer right, Integer bottom) {
    helper.padding(left, top, right, bottom);
    return self();
  }

  @Override
  public GAutoCompleteTextView margin(Integer left, Integer top, Integer right, Integer bottom) {
    helper.margin(left, top, right, bottom);
    return self();
  }
}

File: ganilib-core/src/main/java/com/gani/lib/ui/view/GAutoCompleteTextView.kt
package com.gani.lib.ui.view

import android.content.Context
import android.support.v7.widget.AppCompatAutoCompleteTextView
import android.util.AttributeSet

class GAutoCompleteTextView : AppCompatAutoCompleteTextView, IView {
    private var helper = ViewHelper(this)

    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)

    private fun self(): GAutoCompleteTextView {
        return this
    }

    override fun width(width: Int?): GAutoCompleteTextView {
        helper.width(width)
        return self()
    }

    override fun height(height: Int?): GAutoCompleteTextView {
        helper.height(height)
        return self()
    }

    override fun padding(l: Int?, t: Int?, r: Int?, b: Int?): GAutoCompleteTextView {
        helper.padding(l, t, r, b)
        return self()
    }

    override fun margin(l: Int?, t: Int?, r: Int?, b: Int?): GAutoCompleteTextView {
        helper.margin(l, t, r, b)
        return self()
    }

    override fun bgColor(res: Int): GAutoCompleteTextView {
        helper.bgColor(res)
        return self()
    }
}
